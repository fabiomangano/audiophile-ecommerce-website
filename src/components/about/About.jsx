import React from 'react';
import { useMediaQuery } from 'react-responsive';
import { LargeScreen } from '../../mediaQueries';
import Marginer from '../marginer/Marginer';
import style from './About.module.scss';

const About = () => {
  const isLargeScreen = useMediaQuery(LargeScreen);

  return (
    <div className={style.container}>
      <div className={style.image} />
      <Marginer
        direction={isLargeScreen ? 'vertical' : 'horizontal'}
        margin={isLargeScreen ? 7.8 : 2}
      />
      <div className={style.content}>
        <h4 className={style.heading}>
          Bringing you the <span className={style.brick_word}>best </span>
          audio gear
        </h4>
        <Marginer direction="horizontal" margin={2} />
        <p className={style.about_us}>
          Located at the heart of New York City, Audiophile is the premier store
          for high end headphones, earphones, speakers, and audio accessories.
          We have a large showroom and luxury demonstration rooms available for
          you to browse and experience a wide range of our products. Stop by our
          store to meet some of the fantastic people who make Audiophile the
          best place to buy your portable audio equipment.
        </p>
      </div>
    </div>
  );
};

export default About;
