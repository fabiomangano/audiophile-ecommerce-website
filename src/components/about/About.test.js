import React from 'react';
import { create } from 'react-test-renderer';
import { Context as ResponsiveContext } from 'react-responsive';
import About from './About';

describe('About', () => {
  it('should render correctly when vw is small or less', () => {
    const tree = create(
      <ResponsiveContext.Provider value={{ width: 480 }}>
        <About />
      </ResponsiveContext.Provider>,
    );
    expect(tree.toJSON()).toMatchSnapshot();
  });

  it('should render correctly when vw is medium or greater', () => {
    const tree = create(
      <ResponsiveContext.Provider value={{ width: 769 }}>
        <About />
      </ResponsiveContext.Provider>,
    );
    expect(tree.toJSON()).toMatchSnapshot();
  });

  it('should render correctly when vw is large', () => {
    const tree = create(
      <ResponsiveContext.Provider value={{ width: 1024 }}>
        <About />
      </ResponsiveContext.Provider>,
    );
    expect(tree.toJSON()).toMatchSnapshot();
  });
});
