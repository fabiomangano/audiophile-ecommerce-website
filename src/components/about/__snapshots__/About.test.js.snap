// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`About should render correctly when vw is large 1`] = `
<div
  className="container"
>
  <div
    className="image"
  />
  <span
    style={
      Object {
        "display": "flex",
        "width": "7.8rem",
      }
    }
  />
  <div
    className="content"
  >
    <h4
      className="heading"
    >
      Bringing you the 
      <span
        className="brick_word"
      >
        best 
      </span>
      audio gear
    </h4>
    <span
      style={
        Object {
          "display": "flex",
          "height": "2rem",
        }
      }
    />
    <p
      className="about_us"
    >
      Located at the heart of New York City, Audiophile is the premier store for high end headphones, earphones, speakers, and audio accessories. We have a large showroom and luxury demonstration rooms available for you to browse and experience a wide range of our products. Stop by our store to meet some of the fantastic people who make Audiophile the best place to buy your portable audio equipment.
    </p>
  </div>
</div>
`;

exports[`About should render correctly when vw is medium or greater 1`] = `
<div
  className="container"
>
  <div
    className="image"
  />
  <span
    style={
      Object {
        "display": "flex",
        "height": "2rem",
      }
    }
  />
  <div
    className="content"
  >
    <h4
      className="heading"
    >
      Bringing you the 
      <span
        className="brick_word"
      >
        best 
      </span>
      audio gear
    </h4>
    <span
      style={
        Object {
          "display": "flex",
          "height": "2rem",
        }
      }
    />
    <p
      className="about_us"
    >
      Located at the heart of New York City, Audiophile is the premier store for high end headphones, earphones, speakers, and audio accessories. We have a large showroom and luxury demonstration rooms available for you to browse and experience a wide range of our products. Stop by our store to meet some of the fantastic people who make Audiophile the best place to buy your portable audio equipment.
    </p>
  </div>
</div>
`;

exports[`About should render correctly when vw is small or less 1`] = `
<div
  className="container"
>
  <div
    className="image"
  />
  <span
    style={
      Object {
        "display": "flex",
        "height": "2rem",
      }
    }
  />
  <div
    className="content"
  >
    <h4
      className="heading"
    >
      Bringing you the 
      <span
        className="brick_word"
      >
        best 
      </span>
      audio gear
    </h4>
    <span
      style={
        Object {
          "display": "flex",
          "height": "2rem",
        }
      }
    />
    <p
      className="about_us"
    >
      Located at the heart of New York City, Audiophile is the premier store for high end headphones, earphones, speakers, and audio accessories. We have a large showroom and luxury demonstration rooms available for you to browse and experience a wide range of our products. Stop by our store to meet some of the fantastic people who make Audiophile the best place to buy your portable audio equipment.
    </p>
  </div>
</div>
`;
